.circleci/config.yml

```diff
@@ -8,34 +7,80 @@ jobs:
 
     steps:
       - checkout
-      # Download and cache dependencies
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum "requirements.txt" }}
-            # fallback to using the latest cache if no exact match is found
-            - v1-dependencies-
-
-      - run:
-          name: install dependencies
-          command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r requirements.txt
       - run:
-          name: build phase0 spec
-          command: make build/phase0
+          name: Build pyspec
+          command: make pyspec
 
-      - save_cache:
-          paths:
-            - ./venv
-          key: v1-dependencies-{{ checksum "requirements.txt" }}
+      - run:
+          name: Run py-tests
+          command: make test
 
       - run:
-          name: run tests
-          command: |
-            . venv/bin/activate
-            pytest tests
+          name: Generate YAML tests
+          command: make gen_yaml_tests
 
       - store_artifacts:
           path: test-reports
           destination: test-reports
+
+      - run:
+          name: Save YAML tests for deployment
+          command: |
+            mkdir /tmp/workspace
+            cp -r yaml_tests /tmp/workspace/
+            git log -1 >> /tmp/workspace/latest_commit_message
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - yaml_tests
+            - latest_commit_message
+  commit:
```

Comment:
Copied this part of the configuration over. "commit" is just a job name afaik (circle is not my CI choice). Already discussed this with Danny for a bit, and requested a look from his side specifically on the circle config, to ensure it works as intended.

The results of the discussion:

We probably don't want to run yaml test generators on every commit/PR. As it just slows things down. Or we can allow that, and see when yaml test gen breaks. We can change it to run on any merge into master or something, or check up on it in some other way. We do want it to build and commit a valid set of yaml tests on release. Alternatively, you would need some funny work around, like some "yaml" branch, where we merge into when we need it to build. Open for suggestions on flow there.

Reply:
> Copied this part of the configuration over. "commit" is just a job name afaik (circle is not my CI choice).

I mean, afaik the current auto-push `eth2-test-generators` -[Bot]-> `eth2.0-tests` is triggered by every merged PR in `eth2.0-test-generators`, I assume that we can update the workflow to by branch or by each release if we want.

Some considerations:
1. IMO It totally makes sense that we only push tests to `tests` when new specs release and we are confident that the pushed tests are correct in format or versions.
2. But if we make it automatically, we *might* lose the space/period to check the basic correctness of these tests. In that case, we will make a new `eth2.0-specs` release patch to fix the test gen.

I guess we won't know it surely before we try it, I just put some thought. 🙂

"yaml" branch also seems doable compromise as a protection since we don't have a middle repo (`eth2.0-test-generators`) now.

Reply:
This yaml looks like all branches from the spec would push commits to `eth2.0-tests`. A workaround is to change the `git push origin master` in Line 68 to `git push origin $CIRCLE_BRANCH`. One benefit is we can have different tests for different branches.

Reply:
Shall we just disable this part of the CI script (comment it), and have CI triggered runs of yaml test generation be a future PR, to keep the scope of this PR manageable?

Reply:
See #928